mitodl/odl-video-service
# Added video delete functionality

#### What are the relevant tickets?
Closes #202 

#### What's this PR do?
Adds a button to delete videos on both the collection detail and video detail pages

#### How should this be manually tested?
Log into the app as an admin for some collection. Delete a video on the collection detail page, then delete another video on the video detail page.

After that, make sure you can't see the same buttons for a non-admin user

#### Where should the reviewer start?
`DeleteVideoDialog.js`

#### Any background context you want to provide?

I also enabled the video detail page link for videos in an error state. This was requested in the issue

#### Screenshots (if appropriate)

<img width="983" alt="ss 2017-09-28 at 09 52 32" src="https://user-images.githubusercontent.com/14932219/30988476-6278bc7c-a468-11e7-845c-13c53d16448d.png">
<img style="border:1px solid black;" width="391" alt="ss 2017-09-28 at 09 52 45" src="https://user-images.githubusercontent.com/14932219/30988483-64ed5bc0-a468-11e7-925b-773e427295ca.png">
<img src="https://user-images.githubusercontent.com/14932219/30988493-73199ab0-a468-11e7-8651-4197e898a74c.png">
